From c8dde02842de8fd94d683c40ad5bbb665a5f8822 Mon Sep 17 00:00:00 2001
From: dmitry pervushin <dpervushin@nvidia.com>
Date: Thu, 17 Nov 2016 13:39:25 +0100
Subject: [PATCH 349/352] mm: slub: reenabling interrupts in PREEMPT_RT

If we are past SYSTEM_BOOTING_SINGLECORE, we have smp fully
initialized and should allow our spinlocks to sleep

Bug 1828392

Signed-off-by: dmitry pervushin <dpervushin@nvidia.com>
Reviewed-on: http://git-master/r/1255582
Reviewed-by: Sumeet Gupta <sumeetg@nvidia.com>
---
 mm/slub.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/slub.c b/mm/slub.c
index e6b6809..933c4b32 100644
--- a/mm/slub.c
+++ b/mm/slub.c
@@ -1562,7 +1562,7 @@ static struct page *allocate_slab(struct kmem_cache *s, gfp_t flags, int node)
 	if (gfpflags_allow_blocking(flags))
 		enableirqs = true;
 #ifdef CONFIG_PREEMPT_RT_FULL
-	if (system_state == SYSTEM_RUNNING)
+	if (system_state > SYSTEM_BOOTING_SINGLECORE)
 		enableirqs = true;
 #endif
 	if (enableirqs)
-- 
2.7.4

